### PR TITLE
feat: `CONTRIBUTING.md`

### PR DESCRIPTION
**Pull request by ten3roberts**
Resolves: #581

### CODE REVIEW

**Comment by CKolkey** [comment]
Great start. I'd be happy to help expand on the architecture section, unless you want to. The only thing I'd add is more insistence on test coverage.. Thats already showing up as a weak spot with fixes breaking things...

**Comment by ten3roberts** [comment]
Yes, I'll add that section.

What is the general concensus in markdown? Should proper nouns such as Neogit and Magit be put in *Neogit* or `Neogit`, or just plain?

**Comment by CKolkey** [comment]
Hmm. I don't have a strong feeling on it. I like the backticks, as it makes it stand out nicely, but I'm pretty open to either.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+[`Plenary`](https://github.com/nvim-lua/plenary.nvim) for logging.
+
+Export the environment variables `NEOGIT_LOG_CONSOLE="sync"` to enable logging, and `NEOGIT_LOG_LEVEL="debug"` for more
+verbose logging.
+
+```lua
+local logger = require("neogit.logger")
+
+logger.fmt_info("This is a log message: %d", 2)
+logger.fmt_debug("This is a verbose log message: %q", status)
```
Comment (CKolkey) [inline CONTRIBUTING.md:60]: it's just `info` and `debug`, `trace`, `warn`- no prefix

**Comment by ten3roberts** [comment]
I think this should be a good enough starting point for the contributing guide, and over time we'll refine and add more sections to it when we find parts to be unintuitive.

**Pull request merged**